packages/core/documentation/src/components/DocumentationDisplay.vue

```diff
@@ -0,0 +1,270 @@
+          :color="KUI_COLOR_TEXT_NEUTRAL"
+        >
+          {{ createdAt }}
+        </KBadge>
+      </div>
+
+      <div
+        class="document-display-actions"
+      >
+        <PermissionsWrapper
+          :auth-function="() => canEdit()"
+        >
+          <KInputSwitch
+            v-if="!hidePublishToggle && !isCard"
+            v-model="publishModel"
+            class="document-publish-toggle"
+            data-testid="document-publish-toggle"
+            :label="publishedStatusText"
+            label-position="left"
+            @click="handlePublishToggle"
+          />
+          <KButton
+            appearance="secondary"
+            class="document-edit-button"
+            data-testid="document-edit-button"
+            size="small"
+            @click="emit('edit-clicked')"
+          >
+            {{ i18n.t('documentation.documentation_display.edit_button') }}
+          </KButton>
+          <KButton
+            appearance="primary"
+            data-testid="add-new-page-button"
+            size="small"
+            @click="emit('add-clicked')"
+          >
+            {{ i18n.t('documentation.documentation_display.add_new') }}
+          </KButton>
+        </PermissionsWrapper>
+      </div>
+    </div>
+
+    <div
+      v-if="isLoading"
+      class="markdown-content-loading"
+      data-testid="markdown-content-loading"
+    >
+      <KSkeleton
+        v-if="!isCard"
+        class="markdown-content-loader"
+      />
+
+      <KSkeleton
+        v-else
+        type="card"
+      />
+    </div>
+
+    <div v-else>
+      <div
+        class="document-content"
+        :class="{ 'content-card-view': isCard }"
+      >
+        <DocumentViewer
+          v-if="defaultDocument"
+          :document="defaultDocument"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+import composables from '../composables'
+import DocumentViewer from '@kong-ui-public/document-viewer'
+import { KUI_COLOR_BACKGROUND_NEUTRAL_WEAKER, KUI_COLOR_TEXT_NEUTRAL } from '@kong/design-tokens'
+import { isObjectEmpty } from '../helpers'
+import '@kong-ui-public/document-viewer/dist/style.css'
+import type { PropType } from 'vue'
+import type { DocumentTree } from '../types'
+
+const props = defineProps({
+  /**
+   * A synchronous or asynchronous function which returns a boolean evaluating
+   * if the user can edit an entity by create a new document
+   */
+  canEdit: {
+    type: Function as PropType<() => boolean | Promise<boolean>>,
+    required: false,
+    default: async () => false,
+  },
+  hidePublishToggle: {
+    type: Boolean,
+    default: false,
+  },
+  isCard: {
+    type: Boolean,
+    default: false,
+  },
+  selectedDocument: {
+    type: Object as PropType<{ document: DocumentTree, ast: Object, status: 'published' | 'unpublished'}>,
+    default: () => null,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'add-clicked'): void,
```

Comment:
Do we need “clicked” for these? Could it just be “add” and “edit”?

Reply:
fixed